pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -100,6 +107,141 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
+	// get a list of all ingress on the cluster
+	ingressControllerList := &operatorv1.IngressControllerList{}
+	listOptions := []client.ListOption{
+		client.InNamespace("openshift-ingress-operator"),
+	}
+	err = r.client.List(context.TODO(), ingressControllerList, listOptions...)
+	if err != nil {
+		log.Error(err, "Cannot get list of ingresscontroller")
+		return reconcile.Result{}, err
+	}
```

Comment:
Note for future us:

In order to support this out-of-namespace access, this operator needs to have its `WATCH_NAMESPACE` envvar set to `` (empty string), or else the `List` operation will silently fail. (The reason it fails is because it will ignore the given namespace and look instead to the value of `WATCH_NAMESPACE`.)